examples/elf_dump/elf_dump.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"bytes"
+	"debug/elf"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/intel/tfortools"
+)
+
+var cfg *tfortools.Config
+var code string //The format string
+
+func init() {
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %s [-f <template>] <filename> \n", os.Args[0])
+		flag.PrintDefaults()
+
+		fmt.Fprintf(os.Stderr, "The template passed to the -%s option operates on a\n\n", "f")
+		fmt.Fprintln(os.Stderr, tfortools.GenerateUsageUndecorated(elf.File{}))
+
+		fmt.Fprintln(os.Stderr, tfortools.TemplateFunctionHelp(cfg))
+	}
+
+	flag.StringVar(&code, "f", "", "string containing the template code to execute")
+
+	cfg = tfortools.NewConfig(tfortools.OptAllFns)
+}
+
+func main() {
+	flag.Parse()
+
+	if len(flag.Args()) < 1 {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	filename := flag.Args()[0]
+	if filename == "" {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	contents, err := ioutil.ReadFile(filename)
```

Comment:
Maybe should just call os.Open and pass the file directly into elf.NewFile.  This would get rid of one line of code and would avoid having to read the whole binary into memory.